connection.go

```diff
@@ -87,41 +87,75 @@ func (sc *snowflakeConn) exec(
 	}
 	logger.WithContext(ctx).Infof("parameters: %v", req.Parameters)
 
-	tsmode := "TIMESTAMP_NTZ"
-	idx := 1
-	if len(bindings) > 0 {
-		req.Bindings = make(map[string]execBindParameter, len(bindings))
-		for i, n := 0, len(bindings); i < n; i++ {
-			t := goTypeToSnowflake(bindings[i].Value, tsmode)
-			logger.WithContext(ctx).Debugf("tmode: %v\n", t)
-			if t == "CHANGE_TYPE" {
-				tsmode, err = dataTypeMode(bindings[i].Value)
-				if err != nil {
-					return nil, err
+	requestID := getOrGenerateRequestIDFromContext(ctx)
+	uploader := bindUploader{
```

Comment:
'uploader' is not needed for other exec branches when there is no bind. This also means that arrayBindValueCount and isArrayBind do not have to be part of binUploader. creation of uploader and below added logic should be put under a 'if' branch to reduce unnecessary work for not-bind-upload workflows.

Reply:
Makes sense.